### PR TITLE
Removed problematic parts of #575

### PR DESCRIPTION
**Pull request by NishidaRyu416**
# Overview

# Test Plan

### CODE REVIEW

File: ios/InstagramStories.m [hunk 1]
```diff
@@ -22,7 +22,7 @@ @implementation InstagramStories
 
 - (void)backgroundImage:(NSData *)backgroundImage attributionURL:(NSString *)attributionURL {
     // Verify app can open custom URL scheme, open if able
-    
+
```
Comment (mikehardy) [inline ios/InstagramStories.m:25]: Nothing in this file changed except whitespace, which unnecessarily slows down review

File: ios/RNShare.m [hunk 2]
```diff
@@ -47,6 +47,12 @@
 #import "FacebookStories.h"
 #import "GooglePlusShare.h"
 #import "EmailShare.h"
+
+
+#import "InstagramStories.h"
+// add snapchatshare
```
Comment (mikehardy) [inline ios/RNShare.m:53]: strange whitespace here and the comment seems unnecessary - all the imports should just go one after the other I think?

File: ios/SnapchatShare.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
+//
+//  GooglePlusShare.h
+//  RNShare
+//
+//  Created by Diseño Uno BBCL on 23-07-16.
+//  Copyright © 2016 Facebook. All rights reserved.
```
Comment (mikehardy) [inline ios/SnapchatShare.h:6]: These aren't right, should be copyright you with license MIT or similar

File: ios/SnapchatShare.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+//
+#import <UIKit/UIKit.h>
+// import RCTConvert
+#if __has_include(<React/RCTConvert.h>)
+#import <React/RCTConvert.h>
+#elif __has_include("RCTConvert.h")
+#import "RCTConvert.h"
+#else
+#import "React/RCTConvert.h"   // Required when used as a Pod in a Swift project
+#endif
```
Comment (mikehardy) [inline ios/SnapchatShare.h:16]: I'm not sure if these are still necessary in 3 different ways? Is it possible to just import directly nowadays?

File: ios/SnapchatShare.m [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        }
+    }
+    return self;
+}
+
+- (void *)shareSingle:(NSDictionary *)options
+    failureCallback:(RCTResponseErrorBlock)failureCallback
+    successCallback:(RCTResponseSenderBlock)successCallback {
+
+    NSLog(@"Try open view");
```
Comment (mikehardy) [inline ios/SnapchatShare.m:36]: I'm not sure if other share options directly log but I don't think that's good practice? At minimum if you do log it should contain enough information to know exactly where to locate the log for problem localization, e.g. 'RNShare::shareSingle - Try open view'

File: package-lock.json [hunk 6]
```diff
@@ -0,0 +1,15982 @@
+{
```
Comment (mikehardy) [inline package-lock.json:1]: I don't believe this file should be included as part of the change?

**Comment by NishidaRyu416** [comment]
> My biggest question on this one is if the Snapchat SDK could be made optional? I'm not sure what this will do to app size for people that don't use Snapchat? I believe for instance facebook share works but without directly including the facebook SDK so I'm not sure why the Snapchat SDK has to be brought in as a direct dependency

We need the SDK because all other sharing has been done with deep linking, which you can grasp by seeing the implementations, however SnapChat needs it because it does not support sharing with deep linking.

File: package-lock.json (hunk 6 above)
Comment (NishidaRyu416) [reply package-lock.json:1]: Maybe #575's author is using npm other than yarn?
Should I include a line to exclude it to .gitignore for future comers?

**Comment by mikehardy** [comment]
Ah - I didn't realize you were picking up the baton from another PR, yeah, that is a thankless job :rofl: - I'm still not sure about the headers being tried in 3 ways - is that really needed? - but cool the SDK can be optional. Not sure if it will still compile correctly if it is optional, the whole implementation might need an IFDEF check for the Snapchat SDK header definition then - but still might be possible

**Comment by NishidaRyu416** [comment]
@mikehardy, where to install modules specifically for the example app?
It seems that rn-fetch-blob is needed

**Comment by mikehardy** [comment]
that would be in example/package.json or example/ios/Podfile I think?

**Comment by NishidaRyu416** [comment]
https://react-native-community.github.io/react-native-share/docs/contributing#where-is-the-packagejson-and-node_modules

and after taking a look at https://github.com/react-native-community/datetimepicker/blob/488bedac6a41ccb8c6b34d2a07824de6fd77f47a/package.json, I guess we should put it in devDependencies.

**Comment by NishidaRyu416** [comment]
Unfortunately, I cannot get the example app worked: snapchat does not launch although I follow the doc.
Maybe the SDK is broken?
I hope somebody who is more familiar with native codes investigate this more:)

**Comment by NishidaRyu416** [comment]
Snap Inc. says they have fixed the SDK so the code might work as expected after updating the SDK!

**Comment by franzejr** [comment]
@NishidaRyu416 what is the version you are currently using ?

**Pull request closed without merging**